app/controllers/search_controller.rb

```diff
@@ -10,43 +10,52 @@ def index
   end
 
   def search
-    # Page.reindex
     @page_title = params[:query] == "*" ? t(:see_more) : params[:query]+ "| "+ t(:search_results)
-    regex = ".*"+params[:query].downcase+".*"
-    # scientific_names_result = ScientificName.search params[:query] do |body|
-      # body[:query] = {
-        # regexp:{
-            # canonical_form: regex
-        # }
-     # }
-    # end
-    # @scientific_names = scientific_names_result.results
-    
-    page_result = Page.search params[:query] do |body|
-      body[:query] = {
-        regexp:{
-            scientific_name: regex
-        }
-     }
-    end
-    @pages = page_result.results
-    #debugger
-    # if params[:scientific_names]
-      # @results = @scientific_names
-    # end
+    regex = ".*" + params[:query].downcase + ".*"
+    page_result_scientific_names = search_pages(regex)
+    page_result_vernaculars = search_vernaculars(regex)
+    @pages = merge_results(page_result_scientific_names, page_result_vernaculars)
+
     if params[:pages]
       @results.nil??  @results= @pages : @results +=@pages
     end
     if (params[:scientific_names].nil? && params[:pages].nil?)
       @results = @pages
     end
     unless @results.empty?
-      # debugger
-      @results = @results.paginate(:page => params[:page], :per_page => ENV['per_page'])
-      # @results =@results.uniq!
+      @results = @results.paginate( page: params[:page], per_page: ENV['per_page'])
     else
       flash[:notice] = t(:no_results) +" "+ params[:query]
       redirect_back(fallback_location: root_path)
     end
   end
+  
+  def search_pages(regex)
+    Page.search params[:query] do |body|
+      body[:query] = {
+        regexp:{
+           name_string: regex
+          }
+             }
+    end
+  end
+  
+  def search_vernaculars(regex)
+    Vernacular.search params[:query] do |body|
+      body[:query] = {
+        regexp:{
+           name_string: regex
+          }
+             }
+    end
+  end
+  
+  def merge_results(page_result_scientific_names, page_result_vernaculars)
+    @pages = Array.new
+    @pages += page_result_scientific_names  
+    @pages += page_result_vernaculars.results.map(&:page).uniq
```

Comment:
what is the difference between "unique" in this line and the line after?

Reply:
For a given page, there can be more than one vernacular matching the search query used, which means it may occur more than once in the results. So, uniq is used for the vernaculars search results.
Unlike the scientific names, where a page can only have one scientific name, which would make it of no use to use uniq method here with the search results of scientific names.